src/components/calcite-date-picker/utils.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { getLocaleData, requestCache, translationCache } from "./utils";
+import { getAssetPath } from "@stencil/core";
+
+function setupFetchStub(data: any) {
+  return function fetchStub(): Promise<any> {
+    return new Promise((resolve) => {
+      resolve({
+        json: () => Promise.resolve({ data })
+      });
+    });
+  };
+}
+
+describe("utils", () => {
+  describe("getLocaleData", () => {
+    const fakeData = { fake: "fake data not meant to be checked" };
+    global.fetch = jest.fn().mockImplementation(setupFetchStub(fakeData));
+
+    it("defaults to en locale if lang code is invalid", async () => {
+      const locale = "american";
+      await getLocaleData(locale);
+      const localeData = await requestCache;
+
+      expect(localeData).toHaveProperty("en");
+      delete localeData.en;
+      jest.fn().mockClear();
+    });
+
+    it("fetches locale with conventional-cased lang code", async () => {
+      const locale = "es";
+      await getLocaleData(locale);
+      const localeData = await requestCache;
+
+      expect(localeData).toHaveProperty("es");
+      delete localeData.es;
```

Comment:
If possible, try to move this and the clearing of the mock into the `afterEach` suggested above.